operators/src/processing/rasterization/mod.rs

```diff
@@ -0,0 +1,980 @@
+    type RasterType = f64;
+
+    async fn raster_query<'a>(
+        &'a self,
+        query: RasterQueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> util::Result<BoxStream<'a, util::Result<RasterTile2D<Self::RasterType>>>> {
+        let tiling_strategy = self
+            .tiling_specification
+            .strategy(query.spatial_resolution.x, -query.spatial_resolution.y);
+
+        let tile_size_x = tiling_strategy.tile_size_in_pixels.axis_size_x();
+        let tile_size_y = tiling_strategy.tile_size_in_pixels.axis_size_y();
+
+        let tiles = stream::iter(tiling_strategy.tile_information_iterator(query.spatial_bounds))
+            .then(move |tile_info| async move {
+                let tile_bounds = tile_info.spatial_partition();
+
+                let vector_query = VectorQueryRectangle {
+                    spatial_bounds: BoundingBox2D::new(
+                        tile_bounds.lower_left() - Coordinate2D::new(self.radius, self.radius),
+                        tile_bounds.upper_right() + Coordinate2D::new(self.radius, self.radius),
+                    )?,
+                    time_interval: query.time_interval,
+                    spatial_resolution: query.spatial_resolution,
+                };
+
+                let tile_geo_transform = tile_info.tile_geo_transform();
+
+                let mut tile_data;
+
+                if let MultiPoint(processor) = &self.input {
+                    let mut chunks = processor.query(vector_query, ctx).await?;
+
+                    tile_data = vec![0.; tile_size_x * tile_size_y];
+
+                    while let Some(chunk) = chunks.next().await {
+                        let chunk = chunk?;
+                        ctx.thread_pool().install(|| {
+                            rayon::scope(|s| {
+                                let (send, receive) = std::sync::mpsc::channel();
+
+                                s.spawn(move |_s1| {
+                                    chunk
+                                        .coordinates()
+                                        .par_iter()
+                                        .filter_map(|&coord| {
+                                            let spatial_partition = SpatialPartition2D::new(
+                                                coord
+                                                    - Coordinate2D::new(self.radius, -self.radius),
+                                                coord
+                                                    + Coordinate2D::new(self.radius, -self.radius),
+                                            )
+                                            .expect("Bounds should be valid since radius was checked to be strictly positive")
+                                            .intersection(&tile_info.spatial_partition());
+
+                                            let grid_bounds = match spatial_partition {
+                                                None => { return None; },
```

Comment:
Maybe add a comment what happens here and why you can return `None`

Reply:
https://github.com/geo-engine/geoengine/pull/683/commits/e5da1695b25feb3f71272a1a504d7ba438abd053

This should not strictly be necessary since I only query for points with distance of at most radius from the tile borders so the intersection should never be empty. However, I added this check to be safe when a vector source returns points further away (there is a comment in the OGR source that the spatial filter can be inaccurate, for example). In case the intersection is actually empty nothing needs to be computed so no output values are produced (that's why I return None in the filter_map call).